pkg/util/tracingservice/service_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tracingservice
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing/tracingpb"
+	"github.com/cockroachdb/cockroach/pkg/util/tracingservice/tracingservicepb"
+	"github.com/gogo/protobuf/types"
+	"github.com/stretchr/testify/require"
+)
+
+func setupTraces(localTracer, remoteTracer *tracing.Tracer) (uint64, uint64) {
+	// Start a root span on "node 1".
+	root := localTracer.StartSpan("root", tracing.WithForceRealSpan())
+	root.SetVerbose(true)
+	root.RecordStructured(&types.StringValue{Value: "root"})
+	// Start a child span on "node 1".
+	child := localTracer.StartSpan("root.child", tracing.WithParentAndAutoCollection(root))
+
+	// Start a remote child span on "node 2".
+	childChild := remoteTracer.StartSpan("root.child.remotechild", tracing.WithParentAndManualCollection(child.Meta()))
+	childChild.RecordStructured(&types.StringValue{Value: "root.child.remotechild"})
+	// Start another remote child span on "node 2" that we finish.
+	childChildFinished := remoteTracer.StartSpan("root.child.remotechilddone", tracing.WithParentAndManualCollection(child.Meta()))
+	child.ImportRemoteSpans(childChildFinished.GetRecording())
+	childChildFinished.Finish()
+
+	// Start a root span on "node 2".
+	root2 := remoteTracer.StartSpan("root2", tracing.WithForceRealSpan())
+	root2.SetVerbose(true)
+	root2.RecordStructured(&types.StringValue{Value: "root2"})
+	// Start a child span on "node 2".
+	child2 := remoteTracer.StartSpan("root2.child", tracing.WithParentAndAutoCollection(root2))
+	// Start a remote child span on "node 1".
+	_ = localTracer.StartSpan("root2.child.remotechild", tracing.WithParentAndManualCollection(child2.Meta()))
+	return root.TraceID(), root2.TraceID()
+}
+
+func checkSpanRecordings(t *testing.T, expected, actual []tracingpb.RecordedSpan) {
+	for i, rec := range expected {
+		actualRec := actual[i]
+		require.Equal(t, rec.TraceID, actualRec.TraceID)
+		require.Equal(t, rec.Operation, actualRec.Operation)
+		require.Equal(t, rec.InternalStructured, actualRec.InternalStructured)
+	}
+}
+
+func TestTraceServiceGetSpanRecordings(t *testing.T) {
```

Comment:
Compared to the other test, I don't think this is doing much. All it's effectively checking is that we're iterating over all the spans we've configured the service with and calling GetRecording on each one. Given that, I don't think we need all this jazz; i.e. a single sub-test, or rather just a top-level test, would do. All it would check is that all span recordings belonging to the same span are collected.

Reply:
makes sense, changed to a single top-level test.